internal/glinstance/host.go

```diff
@@ -63,6 +63,9 @@ func APIEndpoint(hostname, protocol string) string {
 		protocol = "https"
 	}
 	if IsSelfHosted(hostname) {
+		if strings.HasPrefix(hostname, "http") {
```

Comment:
This isn't a practical operation taking URLs like [httpie.io](https://httpie.io) and others with `http` prefix as part of the hostname. You can use [`utils.IsValidURL(hostname)`](https://github.com/profclems/glab/blob/6e21a963a9af9ce0ea76bc7db951e3df22db4419/pkg/utils/utils.go#L141) found in our [utils](https://github.com/profclems/glab/blob/6e21a963a9af9ce0ea76bc7db951e3df22db4419/pkg/utils/utils.go#L141) package.

Reply:
This was just a dirty fix I so I can display better the issues I encountred. 

I wrote up comment about this https://github.com/profclems/glab/issues/592#issuecomment-807012822 
TDLR: This would fix the problem of https://https://hostname, but config resolution would break since we are storing configs with hostnames without the protocol

Reply:
I change the approach of fixing this completely so this is not relevant anymore.